### PR TITLE
IBC Firewall

### PR DESCRIPTION
**Pull request by bigs**

### CODE REVIEW

File: panic.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
+diff --git a/x/dungeon/ibc_middleware.go b/x/dungeon/ibc_middleware.go
```
Comment (ash-burnt) [inline panic.patch:1]: this file needs to go

File: x/dungeon/ibc_middleware.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package dungeon
```
Comment (ash-burnt) [inline x/dungeon/ibc_middleware.go:1]: this directory needs to go

File: app/app.go [hunk 3]
```diff
@@ -120,6 +120,7 @@ import (
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 
 	appparams "github.com/burnt-labs/burnt/app/params"
+	"github.com/burnt-labs/burnt/x/dungeon"
```
Comment (ash-burnt) [inline app/app.go:123]: this needs to go

File: app/app.go [hunk 4]
```diff
@@ -597,6 +598,7 @@ func NewWasmApp(
 	var transferStack porttypes.IBCModule
 	transferStack = transfer.NewIBCModule(app.TransferKeeper)
 	transferStack = ibcfee.NewIBCMiddleware(transferStack, app.IBCFeeKeeper)
+	transferStack = dungeon.NewIBCMiddleware(transferStack, app.IBCFeeKeeper.Chann, &app.StakingKeeper)
```
Comment (ash-burnt) [inline app/app.go:601]: delete delete delete

**Review by ash-burnt — APPROVED** [review]

**Review by Peartes — APPROVED** [review]
LGTM

**Pull request merged**